admin_sdk/groups_settings/detect_external_access.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+print_group_settingss all the groups in the domain which have 'external' to the domain access.
+Also print_group_settingss their access settings.
+"""
+from __future__ import print_function
+import httplib2
+
+from apiclient import discovery, errors
+from oauth2client import client, tools
+from oauth2client.file import Storage
+
+# If modifying these scopes, delete your previously saved credentials
+# at credentials.json
+SCOPES = ['https://www.googleapis.com/auth/admin.directory.group',
+          'https://www.googleapis.com/auth/apps.groups.settings']
+CLIENT_SECRET_FILE = 'client_secret.json'
+APPLICATION_NAME = 'List Groups with external access'
+
+# External access for view group
+ANYONE_CAN_VIEW_GROUP = 'ANYONE_CAN_VIEW'
+
+# External access for who can join
+ANYONE_CAN_JOIN_GROUP = 'ANYONE_CAN_JOIN'
+
+# Group can have external members
+EXTERNAL_MEMBERS_CAN_JOIN = 'true'
+
+# External access for who can post messages
+ANYONE_CAN_POST_MESSAGE = 'ANYONE_CAN_POST'
+
+# External access for who can view members
+ANYONE_CAN_VIEW_MEMBERSHIP = 'ANYONE_CAN_VIEW'
+
+def get_credentials():
+    """Gets valid user credentials from storage.
```

Comment:
`"""` should be on it's own line if we're being consistent.

Reply:
Done